src/calendar/gui/pickers/DatePicker.ts

```diff
@@ -89,28 +89,44 @@ export class DatePicker implements Component<DatePickerAttrs> {
 				label,
 				helpLabel: () => this.renderHelpLabel(date, nullSelectionText ?? null),
 				disabled,
+				hasPopup: AriaPopupType.Dialog,
 				oninput: (text) => this.handleInput(text, onDateSelected),
-				onfocus: () => {
-					this.showingDropdown = true
+				onfocus: (_, input) => {
+					if (!disabled) {
+						this.showingDropdown = true
+					}
+
+					if (this.domInput == null) {
```

Comment:
This is because we need to grab the `<input>` in order to focus on it. Setting `domInput` in `oncreate()` grabs the wrapping `<div>` which does nothing if we try to focus it.

Reply:
Ah I see now!

`TextFieldAttrs` has `onDomInputCreated()`, perhaps we should use that to make sure that we get the DOM reference as soon as possible? (I didn't check but I think it does what we need) Probably also makes the intent more clear.

We probably should also get rid of empty `oncreate()`.

Reply:
Ah, I did not notice `onDomInputCreated()`! That is a better way to do it, I have pushed again now.